src/components/CurrentWeatherData/index.js

```diff
@@ -1,20 +1,244 @@
-import React from "react";
-import styled from "styled-components";
+import React, { useEffect, useState } from 'react';
+// Libs
+import { useDispatch, useSelector } from 'react-redux';
+import { Trans, useTranslation } from 'react-i18next';
+import { formatDistanceToNow } from 'date-fns';
+// Constants
+import { getWeatherIcon } from '../../constants/iconMapping';
+import { langMap } from '../../constants/language';
+// Redux
+import { fetchWeather, setLastUpdate } from '../../store/slices/weatherSlice';
+// Icons
+import { MdRefresh } from 'react-icons/md';
+// SVG
+import { Wave1, Wave2 } from './Waves';
+// Styles
+import styled from 'styled-components';
 
 export function CurrentWeatherData() {
-  return <Container>Current Wather Data</Container>;
+  const { t, i18n } = useTranslation();
+  const dispatch = useDispatch();
+  const { lastUpdate, currentForecast, lat, lon } = useSelector(
+    (state) => state.weather
+  );
+  const defaultTime = formatDistanceToNow(new Date(), {
+    includeSeconds: true,
+    locale: langMap[i18n.language],
+  });
+  const [updateDate, setUpdateDate] = useState(defaultTime);
+
+  const [fetching, setFetching] = useState(false);
+
+  useEffect(() => {
+    const loop = setInterval(() => {
+      setUpdateDate(
+        formatDistanceToNow(new Date(JSON.parse(lastUpdate)), {
+          includeSeconds: true,
+          locale: langMap[i18n.language],
+        })
```

Comment:
These options passed to `formatDistanceToNow` are the same as above, maybe you can create a util function which encapsulates them and you just pass the date value. Then it can be reused.